scalalib/src/mill/scalalib/PublishModule.scala

```diff
@@ -306,22 +314,44 @@ object PublishModule extends ExternalModule {
     )
   }
 
-  private[scalalib] def checkSonatypeCreds(sonatypeCreds: String): Task[String] = T.task {
-    if (sonatypeCreds.isEmpty) {
-      (for {
-        username <- T.env.get("SONATYPE_USERNAME")
-        password <- T.env.get("SONATYPE_PASSWORD")
-      } yield {
-        Result.Success(s"$username:$password")
-      }).getOrElse(
-        Result.Failure(
-          "Consider using SONATYPE_USERNAME/SONATYPE_PASSWORD environment variables or passing `sonatypeCreds` argument"
-        )
-      )
+  private[mill] def getFinalGpgArgs(initialGpgArgs: String): Seq[String] = {
```

Comment:
If this is to be used from derived classes, `protected` is better than `private[mill]`. If it's API useful outside of this module, just make it public.

Reply:
This is used in non-derived classes in other modules  (e.g. in `SonatypeCentralPublishModule.publishAll`), but I don't think that it makes sense to expose it to outside users, as it's just for internal parsing of arguments and only relevant to the Sonatype/Nexus world. I've left it as `private [mill]`, especially to give later contributors the freedom to refactor it without breaking bincompat.

Reply:
We can leave it for now.